src/internal/attr.c

```diff
@@ -61,21 +61,27 @@ r_obj* node_names(r_obj* x) {
 }
 
 r_obj* ffi_set_names(r_obj* x, r_obj* mold, r_obj* nm, r_obj* env) {
-  int n_kept = 0;
-
-  r_obj* dots = KEEP_N(rlang_dots(env), &n_kept);
```

Comment:
I moved these after the early exit for `nm == NULL` because they aren't needed, just for maximal performance and code clarity